public/app/features/alerting/unified/api/alertRuleApi.ts

```diff
@@ -242,6 +242,7 @@ export const alertRuleApi = alertingApi.injectEndpoints({
         const { path, params } = rulerUrlBuilder(rulerConfig).namespace(namespace);
         return { url: path, params };
       },
+      providesTags: (_result, _error, { namespace }) => [{ type: 'RuleNamespace', id: namespace }],
```

Comment:
This one is important; prior to this change any calls to `upsertRuleGroup` would not invalidate the `rulerNamespace` endpoints